boa/deployments.py

```diff
@@ -123,15 +129,19 @@ def insert_deployment(self, deployment: Deployment):
 
     def _get_deployments_from_sql(self, sql_query: str, parameters=(), /):
         cur = self.db.execute(sql_query, parameters)
-        ret = [Deployment.from_sql_tuple(item) for item in cur.fetchall()]
-        return ret
+        return (Deployment.from_sql_tuple(item) for item in cur)
 
     def _get_fieldnames_str(self) -> str:
         return ",".join(field.name for field in fields(Deployment))
 
-    def get_deployments(self) -> list[Deployment]:
+    def get_deployments_iter(self) -> Generator[Deployment]:
```

Comment:
```suggestion
    def get_deployments_iter(self) -> Generator[Deployment, None, None]:
```

I'm getting a type error trying this code as-is:

```
  File "/path/code/moccasin/.venv/lib/python3.12/site-packages/boa/deployments.py", line 137, in DeploymentsDB
    def get_deployments_iter(self) -> Generator[Deployment]:
                                      ~~~~~~~~~^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.6/Frameworks/Python.framework/Versions/3.12/lib/python3.12/typing.py", line 398, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.6/Frameworks/Python.framework/Versions/3.12/lib/python3.12/typing.py", line 1482, in __getitem__
    _check_generic(self, params, self._nparams)
  File "/path/code/moccasin/.venv/lib/python3.12/site-packages/typing_extensions.py", line 2947, in _check_generic
    raise TypeError(f"Too {'many' if alen > elen else 'few'} arguments"
TypeError: Too few arguments for typing.Generator; actual 1, expected 3
```